src/main_menu.cpp

```diff
@@ -243,6 +242,52 @@ std::vector<std::string> main_menu::load_file( const std::string &path,
     return result;
 }
 
+holiday main_menu:: get_holiday_from_time()
+{
+    std::tm *local_time;
+
+    std::time_t current_time = std::time( nullptr );
+    local_time = std::localtime( &current_time );
+
+    int month = local_time->tm_mon + 1;
+    int day = local_time->tm_mday;
+    int wday = local_time->tm_wday;
+    int year = local_time->tm_year + 1900;
+
+    /*Calculate date of Easter*/
+    int e_month;
+    int e_day;
+
+    int e_a = year % 19;
+    int e_b = year >> 2;
+    int e_c = ( e_b / 25 ) + 1;
+    int e_d = ( e_c * 3 ) >> 2;
+    int e_e = ( ( e_c << 3 ) + 5 ) / 25;
+    int e_f = ( e_a * 19 + e_d - e_e + 15 ) % 30;
+    int e_g = e_f + ( ( 29578 - e_a - ( e_f << 5 ) ) >> 10 );
+    int e_h = e_g - ( ( year + e_b - e_d + e_g + 2 ) % 7 );
+    int e_i = e_h >> 5; // is it April?
+    e_month = e_i + 3;
+    e_day = ( e_h & 31 ) + e_i;
```

Comment:
Mind documenting what's going on here?

Reply:
It's an implementation of Gauss's Easter Algorithm I found, but to be honest, I don't follow all the ins and outs of it myself; the explanations I've found are rather lengthy: https://math.stackexchange.com/questions/896954/decoding-gauss-easter-algorithm.  That's why I did make sure to test the Easter dates out to 2030.

Reply:
Could you explain that in a comment, and possibly provide the same link?

Reply:
Sure.  I wasn't sure we could link websites from source comments.

Reply:
so to add on to this, if at all possible (and you know what they mean) can you make the variable names a little more verbose? and also make `const` the ones that do not change after initiation.

Reply:
Did most of those; although I could not elaborate on the variable names, I used an implementation of the formula that is much more clearly comparable to the source algorithm.